### ISSUE TITLE
ghc 8.8.1-alpha-2 cannot compile Setup.hs

### ISSUE THREAD

**Issue by peti** [simonmar/happy#143]
~~~
$ ghc --make -no-user-package-db Setup
[1 of 1] Compiling Main             ( Setup.lhs, Setup.o )

Setup.lhs:48:22: error:
    Variable not in scope:
      rawSystemProgramConf
        :: Distribution.Verbosity.Verbosity -> t -> ProgramDb -> t1
   |
48 |   let runProgram p = rawSystemProgramConf (fromFlagOrDefault normal (buildVerbosity flags))
   |                      ^^^^^^^^^^^^^^^^^^^^
~~~

**Comment by peti** [comment]
I see now that the Cabal file specifies

~~~
custom-setup
  setup-depends: Cabal <2.6,
                 base >=4.6 && <5,
                 directory <1.4,
                 filepath <1.5
~~~

which would probably avoid that error. The only thing I'm wondering, though, is how to set up such an environment with ghc-8.8.x without requiring a complex tool like `cabal-install`, which I'm pretty sure I can't even bootstrap without having `happy` first.

**Comment by simonmar** [comment]
Is this fixed by #142?

**Comment by peti** [comment]
Yes, the build succeeds fine with those patches applied: https://build.opensuse.org/package/live_build_log/devel:languages:haskell:ghc-8.8.x/happy/openSUSE_Tumbleweed/x86_64
